tests/apps/pubsub-publisher/app.go

```diff
@@ -94,9 +94,15 @@ func performPublish(w http.ResponseWriter, r *http.Request) {
 	}
 	defer postResp.Body.Close()
 
-	log.Printf("Publish succeeded")
-	w.WriteHeader(http.StatusOK)
-	resp = appResponse{Message: "Success"}
+	// pass on status code to the calling application
+	w.WriteHeader(postResp.StatusCode)
+
+	if postResp.StatusCode == http.StatusOK {
+		log.Printf("Publish succeeded")
+		resp = appResponse{Message: "Success"}
+	} else {
+		resp = appResponse{Message: "Failed"}
```

Comment:
Do we not want to print "Publish failed" here?

Reply:
done